Firebase/Auth/Source/FIRAuth.m

```diff
@@ -616,16 +616,16 @@ - (void)signInWithGameCenterCredential:(FIRGameCenterAuthCredential *)credential
                               }];
 }
 
-/** @fn internalSignInWithEmail:link:completion:
+/** @fn internalSignInAndRetrieveDataWithEmail:link:completion:
     @brief Signs in using an email and email sign-in link.
     @param email The user's email address.
     @param link The email sign-in link.
     @param callback A block which is invoked when the sign in finishes (or is cancelled.) Invoked
         asynchronously on the global auth work queue in the future.
  */
-- (void)internalSignInWithEmail:(nonnull NSString *)email
-                           link:(nonnull NSString *)link
-                       callback:(nullable FIRAuthResultCallback)callback {
+- (void)internalSignInAndRetrieveDataWithEmail:(nonnull NSString *)email
+                                          link:(nonnull NSString *)link
+                                      callback:(nullable FIRAuthDataResultCallback)callback {
```

Comment:
A good practice is always checking for callback before calling it. This function call it in a few places without testing if it is null.